### PR TITLE
Added tests related to Plugin Robustness

### PR DESCRIPTION
**Pull request by sandeep-vedam**
Following are the tests that are added in this PR : 
Compositor_Robustness_Test_001.test.js 
DhcpServer_Robustness_Test_001.test.js 
IoConnector_Robustness_Test_001.test.js 
LocationSync_Robustness_Test_001.test.js 
OCDM_Robustness_Test_001.test.js 
TimeSync_Robustness_Test_001.test.js 
VolumeControl_Robustness_Test_001.test.js 
WebServer_Robustness_Test_001.test.js

### CODE REVIEW

File: src/tests/compositor/Compositor_Robustness_Test_001.test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        },
+        {
+          description: 'Get CPU load',
+          test: getCpuLoad,
+          params: constants.deviceInfo,
+          validate() {
+            let cpuload = this.$data.read('cpuload')
+            this.$log('Cpu Load is', cpuload)
+            if (cpuload > 90) {
+              this.$log('CPU load is greater than 90')
```
Comment (wouterlucas) [inline src/tests/compositor/Compositor_Robustness_Test_001.test.js]: Let's throw an error if its false with the same this.$log

File: src/tests/compositor/Compositor_Robustness_Test_001.test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          test: pluginDeactivate,
+          params: constants.compositorPlugin,
+          assert: 'deactivated',
+        },
+        {
+          description: 'Activate Compositor Plugin and check if it is activated',
+          test: pluginActivate,
+          params: constants.compositorPlugin,
+          assert: 'activated',
+        },
```
Comment (wouterlucas) [inline src/tests/compositor/Compositor_Robustness_Test_001.test.js:29]: Should we ask the controller for the list of active plugins and see if it is actually activated?

File: src/tests/dhcpserver/DhcpServer_Robustness_Test_001.test.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+          test: pluginDeactivate,
+          params: constants.dhcpserver,
+          assert: 'deactivated',
+        },
+        {
+          description: 'Activate DHCP Server Plugin and check if it is activated',
+          test: pluginActivate,
+          params: constants.dhcpserver,
+          assert: 'activated',
+        },
```
Comment (wouterlucas) [inline src/tests/dhcpserver/DhcpServer_Robustness_Test_001.test.js:29]: Same here, should we get the list of plugins and check the state?

File: src/tests/dhcpserver/DhcpServer_Robustness_Test_001.test.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        {
+          description: 'Get CPU load',
+          test: getCpuLoad,
+          params: constants.deviceInfo,
+          validate() {
+            let cpuload = this.$data.read('cpuload')
+            this.$log('Cpu Load is', cpuload)
+            if (cpuload > 90) {
+              this.$log('CPU load is greater than 90')
```
Comment (wouterlucas) [inline src/tests/dhcpserver/DhcpServer_Robustness_Test_001.test.js]: Throw error

File: src/tests/ioconnector/IoConnector_Robustness_Test_001.test.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+          test: pluginDeactivate,
+          params: constants.ioconnector,
+          assert: 'deactivated',
+        },
+        {
+          description: 'Activate IOConnector Plugin and check if it is activated',
+          test: pluginActivate,
+          params: constants.ioconnector,
+          assert: 'activated',
+        },
```
Comment (wouterlucas) [inline src/tests/ioconnector/IoConnector_Robustness_Test_001.test.js:29]: Same here

File: src/tests/ioconnector/IoConnector_Robustness_Test_001.test.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        {
+          description: 'Get CPU load',
+          test: getCpuLoad,
+          params: constants.deviceInfo,
+          validate() {
+            let cpuload = this.$data.read('cpuload')
+            this.$log('Cpu Load is', cpuload)
+            if (cpuload > 90) {
+              this.$log('CPU load is greater than 90')
+              return false
```
Comment (wouterlucas) [inline src/tests/ioconnector/IoConnector_Robustness_Test_001.test.js]: Throw error

File: src/tests/locationsync/LocationSync_Robustness_Test_001.test.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+          test: pluginDeactivate,
+          params: constants.locationSyncPlugin,
+          assert: 'deactivated',
+        },
+        {
+          description: 'Activate Location Sync Plugin and check if it is activated',
+          test: pluginActivate,
+          params: constants.locationSyncPlugin,
+          assert: 'activated',
+        },
```
Comment (wouterlucas) [inline src/tests/locationsync/LocationSync_Robustness_Test_001.test.js:29]: Same here

File: src/tests/locationsync/LocationSync_Robustness_Test_001.test.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        {
+          description: 'Get CPU load',
+          test: getCpuLoad,
+          params: constants.deviceInfo,
+          validate() {
+            let cpuload = this.$data.read('cpuload')
+            this.$log('Cpu Load is', cpuload)
+            if (cpuload > 90) {
+              this.$log('CPU load is greater than 90')
+              return false
```
Comment (wouterlucas) [inline src/tests/locationsync/LocationSync_Robustness_Test_001.test.js]: Throw error

File: src/tests/ocdm/OCDM_Robustness_Test_001.test.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+          test: pluginDeactivate,
+          params: constants.ocdmPlugin,
+          assert: 'deactivated',
+        },
+        {
+          description: 'Activate OCDM Plugin and check if it is activated',
+          test: pluginActivate,
+          params: constants.ocdmPlugin,
+          assert: 'activated',
+        },
```
Comment (wouterlucas) [inline src/tests/ocdm/OCDM_Robustness_Test_001.test.js:28]: Same here

File: src/tests/ocdm/OCDM_Robustness_Test_001.test.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        {
+          description: 'Get CPU load',
+          test: getCpuLoad,
+          params: constants.deviceInfo,
+          validate() {
+            let cpuload = this.$data.read('cpuload')
+            this.$log('Cpu Load is', cpuload)
+            if (cpuload > 90) {
+              this.$log('CPU load is greater than 90')
+              return false
```
Comment (wouterlucas) [inline src/tests/ocdm/OCDM_Robustness_Test_001.test.js]: Throw error

File: src/tests/timesync/TimeSync_Robustness_Test_001.test.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+          test: pluginDeactivate,
+          params: constants.timeSyncPlugin,
+          assert: 'deactivated',
+        },
+        {
+          description: 'Activate TimeSync Plugin and check if it is activated',
+          test: pluginActivate,
+          params: constants.timeSyncPlugin,
+          assert: 'activated',
+        },
```
Comment (wouterlucas) [inline src/tests/timesync/TimeSync_Robustness_Test_001.test.js:28]: Same here

File: src/tests/timesync/TimeSync_Robustness_Test_001.test.js [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+        {
+          description: 'Get CPU load',
+          test: getCpuLoad,
+          params: constants.deviceInfo,
+          validate() {
+            let cpuload = this.$data.read('cpuload')
+            this.$log('Cpu Load is', cpuload)
+            if (cpuload > 90) {
+              this.$log('CPU load is greater than 90')
+              return false
```
Comment (wouterlucas) [inline src/tests/timesync/TimeSync_Robustness_Test_001.test.js]: Throw error

File: src/tests/volumecontrol/VolumeControl_Robustness_Test_001.test.js [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+          test: pluginDeactivate,
+          params: constants.volumeControl,
+          assert: 'deactivated',
+        },
+        {
+          description: 'Activate VolumeControl Plugin and check if it is activated',
+          test: pluginActivate,
+          params: constants.volumeControl,
+          assert: 'activated',
+        },
```
Comment (wouterlucas) [inline src/tests/volumecontrol/VolumeControl_Robustness_Test_001.test.js:29]: Same here

File: src/tests/volumecontrol/VolumeControl_Robustness_Test_001.test.js [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+        {
+          description: 'Get CPU load',
+          test: getCpuLoad,
+          params: constants.deviceInfo,
+          validate() {
+            let cpuload = this.$data.read('cpuload')
+            this.$log('Cpu Load is', cpuload)
+            if (cpuload > 90) {
+              this.$log('CPU load is greater than 90')
+              return false
```
Comment (wouterlucas) [inline src/tests/volumecontrol/VolumeControl_Robustness_Test_001.test.js]: Throw error

File: src/tests/webserver/WebServer_Robustness_Test_001.test.js [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+          test: pluginDeactivate,
+          params: constants.webServerPlugin,
+          assert: 'deactivated',
+        },
+        {
+          description: 'Activate WebServer Plugin and check if it is activated',
+          test: pluginActivate,
+          params: constants.webServerPlugin,
+          assert: 'activated',
+        },
```
Comment (wouterlucas) [inline src/tests/webserver/WebServer_Robustness_Test_001.test.js:28]: Same here

File: src/tests/webserver/WebServer_Robustness_Test_001.test.js [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+        {
+          description: 'Get CPU load',
+          test: getCpuLoad,
+          params: constants.deviceInfo,
+          validate() {
+            let cpuload = this.$data.read('cpuload')
+            this.$log('Cpu Load is', cpuload)
+            if (cpuload > 90) {
+              this.$log('CPU load is greater than 90')
+              return false
```
Comment (wouterlucas) [inline src/tests/webserver/WebServer_Robustness_Test_001.test.js]: Throw error

**Review by wouterlucas — CHANGES_REQUESTED** [review]

File: src/tests/compositor/Compositor_Robustness_Test_001.test.js (hunk 2 above)
Comment (sandeep-vedam) [reply src/tests/compositor/Compositor_Robustness_Test_001.test.js:29]: 'pluginActivate' activates the plugin and also **returns the state of the plugin** and it is asserted **using the state as 'activated'**. I think no need of getting the list of active plugins from controller. Please let me know if we can still do as you mentioned.

File: src/tests/dhcpserver/DhcpServer_Robustness_Test_001.test.js (hunk 3 above)
Comment (sandeep-vedam) [reply src/tests/dhcpserver/DhcpServer_Robustness_Test_001.test.js:29]: 'pluginActivate' activates the plugin and also **returns the state of the plugin** and it is asserted **using the state as 'activated'**. I think no need of getting the list of active plugins from controller. Please let me know if we can still do as you mentioned.

File: src/tests/compositor/Compositor_Robustness_Test_001.test.js (hunk 1 above)
Comment (sandeep-vedam) [reply src/tests/compositor/Compositor_Robustness_Test_001.test.js]: Agree and will update the same

File: src/tests/dhcpserver/DhcpServer_Robustness_Test_001.test.js (hunk 4 above)
Comment (sandeep-vedam) [reply src/tests/dhcpserver/DhcpServer_Robustness_Test_001.test.js]: Agree. Will update as per the comment

File: src/tests/ioconnector/IoConnector_Robustness_Test_001.test.js (hunk 5 above)
Comment (sandeep-vedam) [reply src/tests/ioconnector/IoConnector_Robustness_Test_001.test.js:29]: 'pluginActivate' activates the plugin and also **returns the state of the plugin** and it is asserted **using the state as 'activated'**. I think no need of getting the list of active plugins from controller. Please let me know if we can still do as you mentioned.

File: src/tests/ioconnector/IoConnector_Robustness_Test_001.test.js (hunk 6 above)
Comment (sandeep-vedam) [reply src/tests/ioconnector/IoConnector_Robustness_Test_001.test.js]: Agree will update the code as per the comment

File: src/tests/locationsync/LocationSync_Robustness_Test_001.test.js (hunk 7 above)
Comment (sandeep-vedam) [reply src/tests/locationsync/LocationSync_Robustness_Test_001.test.js:29]: 'pluginActivate' activates the plugin and also **returns the state of the plugin** and it is asserted **using the state as 'activated'**. I think no need of getting the list of active plugins from controller. Please let me know if we can still do as you mentioned.

File: src/tests/locationsync/LocationSync_Robustness_Test_001.test.js (hunk 8 above)
Comment (sandeep-vedam) [reply src/tests/locationsync/LocationSync_Robustness_Test_001.test.js]: Agree. Will update the code as per the comment

File: src/tests/ocdm/OCDM_Robustness_Test_001.test.js (hunk 9 above)
Comment (sandeep-vedam) [reply src/tests/ocdm/OCDM_Robustness_Test_001.test.js:28]: 'pluginActivate' activates the plugin and also **returns the state of the plugin** and it is asserted **using the state as 'activated'**. I think no need of getting the list of active plugins from controller. Please let me know if we can still do as you mentioned.

File: src/tests/ocdm/OCDM_Robustness_Test_001.test.js (hunk 10 above)
Comment (sandeep-vedam) [reply src/tests/ocdm/OCDM_Robustness_Test_001.test.js]: Agree. Will update as per the comment

File: src/tests/timesync/TimeSync_Robustness_Test_001.test.js (hunk 11 above)
Comment (sandeep-vedam) [reply src/tests/timesync/TimeSync_Robustness_Test_001.test.js:28]: 'pluginActivate' activates the plugin and also **returns the state of the plugin** and it is asserted **using the state as 'activated'**. I think no need of getting the list of active plugins from controller. Please let me know if we can still do as you mentioned.

File: src/tests/timesync/TimeSync_Robustness_Test_001.test.js (hunk 12 above)
Comment (sandeep-vedam) [reply src/tests/timesync/TimeSync_Robustness_Test_001.test.js]: Agree will update as per the comment

File: src/tests/volumecontrol/VolumeControl_Robustness_Test_001.test.js (hunk 13 above)
Comment (sandeep-vedam) [reply src/tests/volumecontrol/VolumeControl_Robustness_Test_001.test.js:29]: 'pluginActivate' activates the plugin and also **returns the state of the plugin** and it is asserted **using the state as 'activated'**. I think no need of getting the list of active plugins from controller. Please let me know if we can still do as you mentioned.

File: src/tests/volumecontrol/VolumeControl_Robustness_Test_001.test.js (hunk 14 above)
Comment (sandeep-vedam) [reply src/tests/volumecontrol/VolumeControl_Robustness_Test_001.test.js]: Agree will update as per the coment

File: src/tests/webserver/WebServer_Robustness_Test_001.test.js (hunk 15 above)
Comment (sandeep-vedam) [reply src/tests/webserver/WebServer_Robustness_Test_001.test.js:28]: 'pluginActivate' activates the plugin and also **returns the state of the plugin** and it is asserted **using the state as 'activated'**. I think no need of getting the list of active plugins from controller. Please let me know if we can still do as you mentioned.

File: src/tests/webserver/WebServer_Robustness_Test_001.test.js (hunk 16 above)
Comment (sandeep-vedam) [reply src/tests/webserver/WebServer_Robustness_Test_001.test.js]: Agree will update as per the comment

**Review by wouterlucas — APPROVED** [review]
LGTM

**Pull request merged**